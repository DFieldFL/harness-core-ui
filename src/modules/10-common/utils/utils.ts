import type { IconName } from '@wings-software/uicore'
import { Scope } from '@common/interfaces/SecretsInterface'
import { ModuleName } from 'framework/types/ModuleName'

interface SetPageNumberProps {
  setPage: (value: React.SetStateAction<number>) => void
  pageItemsCount?: number
  page: number
}

export const getModuleIcon = (module: ModuleName): IconName => {
  switch (module) {
    case ModuleName.CD:
      return 'cd-main'
    case ModuleName.CV:
      return 'cv-main'
    case ModuleName.CI:
      return 'ci-main'
    case ModuleName.CE:
      return 'ce-main'
    case ModuleName.CF:
      return 'cf-main'
  }
  return 'nav-project'
}

export const getReference = (scope?: Scope, identifier?: string): string | undefined => {
  switch (scope) {
    case Scope.PROJECT:
      return identifier
    case Scope.ORG:
      return `org.${identifier}`
    case Scope.ACCOUNT:
      return `account.${identifier}`
  }
}

export const setPageNumber = ({ setPage, pageItemsCount, page }: SetPageNumberProps): void => {
  if (pageItemsCount === 0 && page > 0) {
    setPage(page - 1)
  }
}
