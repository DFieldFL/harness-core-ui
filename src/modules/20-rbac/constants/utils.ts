export enum RbacResourceGroupTypes {
  DYNAMIC_RESOURCE_SELECTOR = 'DynamicResourceSelector'
}
