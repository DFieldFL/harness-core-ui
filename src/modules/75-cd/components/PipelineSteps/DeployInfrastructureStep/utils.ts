/*
 * Copyright 2022 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import { FormikErrors, yupToFormErrors } from 'formik'
import { defaultTo, get, isEmpty } from 'lodash-es'
import * as Yup from 'yup'

import {
  getMultiTypeFromValue,
  MultiTypeInputType,
  RUNTIME_INPUT_VALUE,
  SelectOption,
  MultiSelectWithSubmenuOption
} from '@harness/uicore'
import type { EnvironmentResponseDTO } from 'services/cd-ng'

import { getDurationValidationSchema } from '@common/components/MultiTypeDuration/MultiTypeDuration'

import type { DeployStageConfig } from '@pipeline/utils/DeployStageInterface'
import { StepViewType, ValidateInputSetProps } from '@pipeline/components/AbstractSteps/Step'

export interface DeployInfrastructureProps {
  initialValues: DeployStageConfig
  onUpdate?: (data: DeployStageConfig) => void
  readonly: boolean
  allowableTypes: MultiTypeInputType[]
  stepViewType?: StepViewType
  inputSetData?: {
    template?: DeployStageConfig
    path?: string
    readonly?: boolean
  }
}

export const ALL_SELECTED = 'All'

export function isEditEnvironment(data?: EnvironmentResponseDTO): boolean {
  if (!isEmpty(data?.identifier)) {
    return true
  }
  return false
}

export function isEditEnvironmentOrEnvGroup(data?: EnvironmentResponseDTO): boolean {
  if (!isEmpty(data?.identifier)) {
    return true
  }
  return false
}

export function isEditInfrastructure(data?: string): boolean {
  if (!isEmpty(data)) {
    return true
  }
  return false
}

export function validateStepForm({
  data,
  template,
  getString,
  viewType
}: ValidateInputSetProps<any>): FormikErrors<any> {
  const isRequired = viewType === StepViewType.DeploymentForm || viewType === StepViewType.TriggerForm
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  const errors = {} as any
  // istanbul ignore next
  // istanbul ignore else
  if (getMultiTypeFromValue(template?.timeout) === MultiTypeInputType.RUNTIME) {
    // istanbul ignore next
    let timeoutSchema = getDurationValidationSchema({ minimum: '10s' })
    // istanbul ignore next
    if (isRequired) {
      // istanbul ignore next
      timeoutSchema = timeoutSchema.required(getString?.('validation.timeout10SecMinimum'))
    }
    const timeout = Yup.object().shape({
      timeout: timeoutSchema
    })

    try {
      timeout.validateSync(data)
    } catch (e) {
      /* istanbul ignore else */
      if (e instanceof Yup.ValidationError) {
        const err = yupToFormErrors(e)

        Object.assign(errors, err)
      }
    }
  }
  return errors
}

export function processNonGitOpsInitialValues(initialValues: DeployStageConfig) {
  return {
    environment: {
      environmentRef: defaultTo(initialValues.environment?.environmentRef, ''),
      deployToAll: defaultTo(initialValues.environment?.deployToAll, false)
    },
    infrastructureRef: (initialValues.environment?.infrastructureDefinitions?.[0].ref ||
      initialValues.environment?.infrastructureDefinitions ||
      '') as string
  }
}

export function processNonGitOpsFormValues(data: DeployStageConfig) {
  return {
    environment: {
      environmentRef: data.environment?.environmentRef,
      deployToAll: data.environment?.environmentRef === RUNTIME_INPUT_VALUE ? true : false,
      ...(data.environment?.environmentRef &&
        data.environment?.environmentRef !== RUNTIME_INPUT_VALUE &&
        data.infrastructureRef && {
          infrastructureDefinitions:
            data.infrastructureRef === RUNTIME_INPUT_VALUE
              ? RUNTIME_INPUT_VALUE
              : [
                  {
                    ref: data.infrastructureRef
                  }
                ]
        })
    }
  }
}

export function processGitOpsEnvironmentInitialValues(initialValues: DeployStageConfig) {
  return {
    isEnvGroup: false,
    environmentOrEnvGroupRef: defaultTo(initialValues.environment?.environmentRef, ''),
    environmentOrEnvGroupAsRuntime: 'Environment',
    deployToAll: defaultTo(initialValues.environment?.deployToAll, false),
    clusterRef:
      getMultiTypeFromValue(initialValues.environment?.gitOpsClusters as any) === MultiTypeInputType.RUNTIME
        ? RUNTIME_INPUT_VALUE
        : initialValues.environment?.deployToAll
        ? [
            {
              label: 'All Clusters in Environment',
              value: ALL_SELECTED
            }
          ]
        : defaultTo(initialValues.environment?.gitOpsClusters, [])?.map(cluster => {
            return {
              label: cluster.ref,
              value: cluster.ref
            }
          })
  }
}

export function processGitOpsEnvironmentFormValues(data: DeployStageConfig) {
  const allClustersSelected = (data.clusterRef as SelectOption[])?.[0].value === ALL_SELECTED

  return {
    environment: {
      environmentRef:
        data.environmentOrEnvGroupRef === RUNTIME_INPUT_VALUE
          ? RUNTIME_INPUT_VALUE
          : defaultTo((data.environmentOrEnvGroupRef as SelectOption)?.value, ''),
      deployToAll: data.environmentOrEnvGroupRef === RUNTIME_INPUT_VALUE ? true : allClustersSelected,
      ...(data.environmentOrEnvGroupRef &&
        data.environmentOrEnvGroupRef !== RUNTIME_INPUT_VALUE &&
        !allClustersSelected && {
          gitOpsClusters:
            data.clusterRef === RUNTIME_INPUT_VALUE
              ? RUNTIME_INPUT_VALUE
              : (data.clusterRef as SelectOption[])?.map(cluster => ({ ref: cluster.value }))
        })
    }
  }
}

export function processGitOpsEnvGroupInitialValues(initialValues: DeployStageConfig) {
  return {
    isEnvGroup: true,
    environmentOrEnvGroupRef: defaultTo(initialValues.environmentGroup?.envGroupRef, ''),
    environmentOrEnvGroupAsRuntime: 'Environment Group',
    environmentsInEnvGroup:
      getMultiTypeFromValue(initialValues.environmentGroup?.envGroupRef) === MultiTypeInputType.RUNTIME
        ? RUNTIME_INPUT_VALUE
        : initialValues.environmentGroup?.deployToAll
        ? [
            {
              name: ALL_SELECTED
            }
          ]
        : defaultTo(initialValues.environmentGroup?.envGroupConfig, [])?.map(environment => {
            return {
              name: environment.environmentRef,
              deployToAll: environment.deployToAll,
              ...(!environment.deployToAll && { clusters: defaultTo(environment.gitOpsClusters, []) })
            }
          }),
    deployToAll: defaultTo(initialValues.environmentGroup?.deployToAll, false),
    clusterRef: initialValues.environmentGroup?.deployToAll
      ? RUNTIME_INPUT_VALUE
      : initialValues.environmentGroup?.envGroupConfig?.reduce((prev, environment) => {
          return [
            ...prev,
            ...(environment.deployToAll
              ? [
                  {
                    label: `All`,
                    value: 'All',
                    parentLabel: environment.environmentRef,
                    parentValue: environment.environmentRef
                  }
                ]
              : defaultTo(
                  environment?.gitOpsClusters?.map(cluster => ({
                    label: cluster.ref,
                    value: cluster.ref,
                    parentLabel: environment.environmentRef,
                    parentValue: environment.environmentRef
                  })),
                  []
                ))
          ]
        }, [] as MultiSelectWithSubmenuOption[])
  }
}

export function processGitOpsEnvGroupFormValues(data: DeployStageConfig) {
  const environmentClusterMap: any = {}
  ;(data.clusterRef as MultiSelectWithSubmenuOption[])?.forEach(cluster => {
    if (cluster.value !== 'All') {
      try {
        environmentClusterMap[get(cluster, 'parentValue', '')].push(cluster.value)
      } catch (e: any) {
        environmentClusterMap[get(cluster, 'parentValue', '')] = [cluster.value]
      }
    }
  })

  const allEnvironmentsSelected = (data.environmentInEnvGroupRef as SelectOption[])?.[0].value === ALL_SELECTED

  return {
    environmentGroup: {
      envGroupRef:
        data.environmentOrEnvGroupRef === RUNTIME_INPUT_VALUE
          ? RUNTIME_INPUT_VALUE
          : defaultTo((data.environmentOrEnvGroupRef as SelectOption)?.value, ''),
      deployToAll: data.environmentOrEnvGroupRef === RUNTIME_INPUT_VALUE || allEnvironmentsSelected,
      ...(!(data.environmentOrEnvGroupRef === RUNTIME_INPUT_VALUE || allEnvironmentsSelected) && {
        envGroupConfig:
          data.environmentInEnvGroupRef === RUNTIME_INPUT_VALUE
            ? RUNTIME_INPUT_VALUE
            : (data?.environmentInEnvGroupRef as SelectOption[])?.map(environmentInEnvGroup => {
                const areClustersSelected = Boolean(environmentClusterMap[environmentInEnvGroup.value])
                return {
                  environmentRef: environmentInEnvGroup?.value,
                  deployToAll: !areClustersSelected,
                  ...(areClustersSelected && {
                    gitOpsClusters:
                      data.clusterRef === RUNTIME_INPUT_VALUE
                        ? RUNTIME_INPUT_VALUE
                        : environmentClusterMap[environmentInEnvGroup.value]?.map((cluster: any) => ({ ref: cluster }))
                  })
                }
              })
      })
    }
  }
}
