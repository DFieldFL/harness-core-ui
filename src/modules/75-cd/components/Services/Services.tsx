import React, { useRef, useState } from 'react'
import { noop } from 'lodash-es'
import { Views, ServiceStoreContext } from './common'

import { ServicesHeader } from './ServicesHeader/ServicesHeader'
import { ServicesContent } from './ServicesContent/ServicesContent'

export const Services: React.FC = () => {
  const [view, setView] = useState(Views.INSIGHT)
  const fetchDeploymentList = useRef<() => void>(noop)
  return (
    <ServiceStoreContext.Provider
      value={{
        view,
        setView,
        fetchDeploymentList
      }}
    >
      <ServicesHeader />
      <ServicesContent />
    </ServiceStoreContext.Provider>
  )
}
