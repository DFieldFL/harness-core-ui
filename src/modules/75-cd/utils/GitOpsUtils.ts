import type { GitOpsInfoDTO, ManagedArgoGitOpsInfoDTO, ConnectedArgoGitOpsInfoDTO } from 'services/cd-ng'

export type GitOpsProviderType = Required<GitOpsInfoDTO>['type']

export const GitOpsProviderTypeEnum: Record<GitOpsProviderType, GitOpsProviderType> = {
  CONNECTED_ARGO_PROVIDER: 'CONNECTED_ARGO_PROVIDER',
  MANAGED_ARGO_PROVIDER: 'MANAGED_ARGO_PROVIDER'
}

export function isConnectedGitOpsProvider(obj?: GitOpsInfoDTO | null): obj is ConnectedArgoGitOpsInfoDTO {
  return obj?.type === GitOpsProviderTypeEnum.CONNECTED_ARGO_PROVIDER
}

export function isManagedGitOpsProvider(obj?: GitOpsInfoDTO | null): obj is ManagedArgoGitOpsInfoDTO {
  return obj?.type === GitOpsProviderTypeEnum.MANAGED_ARGO_PROVIDER
}
