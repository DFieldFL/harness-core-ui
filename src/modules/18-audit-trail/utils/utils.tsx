import React from 'react'
import { Menu } from '@blueprintjs/core'
import type { ItemRenderer } from '@blueprintjs/select'
import { Layout, Text, Avatar, Color, MultiSelectOption } from '@wings-software/uicore'
import { isEmail } from '@common/utils/Validation'

export interface UserItem extends MultiSelectOption {
  email?: string
}

export const UserItemRenderer: ItemRenderer<UserItem> = (item, { handleClick }) => (
  <Menu.Item
    key={item.value.toString()}
    text={
      <Layout.Horizontal spacing="small" flex={{ alignItems: 'center', justifyContent: 'flex-start' }}>
        <Avatar name={item.label} email={item.email || item.value.toString()} size="small" hoverCard={false} />
        <Layout.Vertical padding={{ left: 'small' }}>
          <Text color={Color.BLACK}>{item.label}</Text>
          <Text color={Color.GREY_700}>{item.email || item.value}</Text>
        </Layout.Vertical>
      </Layout.Horizontal>
    }
    onClick={handleClick}
  />
)

export const UserTagRenderer = (item: UserItem, validate = false): React.ReactNode => (
  <Layout.Horizontal key={item.value.toString()} flex spacing="small">
    <Avatar name={item.label} email={item.value.toString()} size="xsmall" hoverCard={false} />
    <Text color={validate && !isEmail(item.value.toString().toLowerCase()) ? Color.RED_500 : Color.BLACK}>
      {item.label}
    </Text>
  </Layout.Horizontal>
)
