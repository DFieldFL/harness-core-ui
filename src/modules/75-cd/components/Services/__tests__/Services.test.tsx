import React from 'react'
import { render } from '@testing-library/react'
import { TestWrapper } from '@common/utils/testUtils'
import { Services } from '../Services'

jest.mock('highcharts-react-official', () => () => <></>)

jest.mock('services/cd-ng', () => {
  return {
    useGetServiceDeploymentsInfo: jest.fn,
    useGetWorkloads: jest.fn,
    useGetServicesGrowthTrend: jest.fn,
    useGetServiceDetails: jest.fn
  }
})

describe('Services', () => {
  test('should render Services', () => {
    const { container } = render(
      <TestWrapper
        path="account/:accountId/cd/orgs/:orgIdentifier/projects/:projectIdentifier/services"
        pathParams={{ accountId: 'dummy', orgIdentifier: 'dummy', projectIdentifier: 'dummy' }}
      >
        <Services />
      </TestWrapper>
    )
    expect(container).toMatchSnapshot()
  })
})
