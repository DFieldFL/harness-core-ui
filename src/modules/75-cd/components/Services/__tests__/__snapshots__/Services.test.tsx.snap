// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Services should render Services 1`] = `
<div>
  <div
    class="Layout--horizontal StyledProps--main container standard StyledProps--flex StyledProps--padding-left-xlarge StyledProps--padding-right-xlarge"
  >
    <div
      class="Layout--vertical StyledProps--main"
    >
      <div
        class="Layout--vertical Layout--layout-spacing-xsmall StyledProps--main"
      >
        <div
          class="breadcrumbs"
        >
          <div>
            <a
              class="breadcrumb"
              href="/account/dummy/undefined/orgs/dummy/projects/dummy/services"
            />
            <span
              class="bp3-icon StyledProps--main separator"
              data-icon="main-chevron-right"
            >
              <svg
                height="11"
                viewBox="0 0 8 14"
                width="11"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M7.7 6.3l-6-6C1.5.1 1.3 0 1 0 .4 0 0 .4 0 1c0 .3.1.5.3.7L5.6 7 .3 12.3c-.2.2-.3.4-.3.7 0 .6.4 1 1 1 .3 0 .5-.1.7-.3l6-6c.2-.2.3-.4.3-.7s-.1-.5-.3-.7z"
                  fill="#4d5969"
                />
              </svg>
            </span>
          </div>
          <div>
            <span
              class="lastBreadcrumb"
            >
              services
            </span>
          </div>
        </div>
        <p
          class="StyledProps--font StyledProps--main StyledProps--font-size-medium StyledProps--color StyledProps--color-grey700"
        >
          services
        </p>
      </div>
    </div>
  </div>
  <div
    class="Layout--horizontal StyledProps--main header StyledProps--flex StyledProps--flex-distribution-space-between"
  >
    <div
      class="Layout--horizontal StyledProps--main"
    >
      <button
        class="bp3-button Button--button StyledProps--font StyledProps--main StyledProps--intent-primary Button--with-current-color"
        data-testid="add-service"
        type="button"
      >
        <span
          class="bp3-icon bp3-icon-plus StyledProps--main StyledProps--padding-right-xsmall"
          icon="plus"
        >
          <svg
            data-icon="plus"
            height="10"
            viewBox="0 0 16 16"
            width="10"
          >
            <desc>
              plus
            </desc>
            <path
              d="M13 7H9V3c0-.55-.45-1-1-1s-1 .45-1 1v4H3c-.55 0-1 .45-1 1s.45 1 1 1h4v4c0 .55.45 1 1 1s1-.45 1-1V9h4c.55 0 1-.45 1-1s-.45-1-1-1z"
              fill-rule="evenodd"
            />
          </svg>
        </span>
        <span
          class="bp3-button-text"
        >
          newService
        </span>
      </button>
    </div>
    <div
      class="Layout--horizontal StyledProps--main StyledProps--flex"
    >
      <button
        class="bp3-button bp3-minimal Button--button StyledProps--font StyledProps--main StyledProps--intent-primary Button--with-current-color Button--without-shadow"
        type="button"
      >
        <span
          class="bp3-icon StyledProps--main"
          data-icon="insight-view"
        >
          <svg
            height="16"
            viewBox="0 0 24 24"
            width="16"
            xmlns="http://www.w3.org/2000/svg"
          >
            <rect
              height="6"
              rx="1"
              width="19"
              x="2"
              y="4"
            />
            <rect
              height="2"
              rx="1"
              width="19"
              x="2"
              y="13"
            />
            <rect
              height="2"
              rx="1"
              width="19"
              x="2"
              y="18"
            />
          </svg>
        </span>
      </button>
      <button
        class="bp3-button bp3-minimal Button--button StyledProps--font StyledProps--main Button--with-current-color Button--without-shadow"
        type="button"
      >
        <span
          class="bp3-icon StyledProps--main"
          data-icon="list-view"
        >
          <svg
            height="16"
            viewBox="0 0 24 24"
            width="16"
            xmlns="http://www.w3.org/2000/svg"
          >
            <rect
              height="2"
              rx="1"
              width="19"
              x="2"
              y="4"
            />
            <rect
              height="2"
              rx="1"
              width="19"
              x="2"
              y="11"
            />
            <rect
              height="2"
              rx="1"
              width="19"
              x="2"
              y="18"
            />
          </svg>
        </span>
      </button>
    </div>
  </div>
  <div
    class="StyledProps--font StyledProps--main pageBody pageBody"
    data-name="page-body"
  >
    <div
      class="Layout--vertical StyledProps--main container StyledProps--margin-left-xlarge StyledProps--margin-right-xlarge StyledProps--margin-top-large StyledProps--margin-bottom-large"
    >
      <div
        class="Layout--horizontal StyledProps--main StyledProps--margin-bottom-large"
      >
        <div
          class="bp3-card bp3-elevation-0 Card--card card"
        >
          <div
            class="Layout--vertical StyledProps--main"
            style="width: 248px;"
          >
            <div
              class="Layout--horizontal StyledProps--main topSection"
            >
              <div
                class="Layout--vertical StyledProps--main"
                style="width: 100%;"
              >
                <p
                  class="StyledProps--font StyledProps--main StyledProps--font-weight-bold StyledProps--color StyledProps--color-grey600"
                >
                  services
                </p>
                <div
                  class="Layout--horizontal StyledProps--main StyledProps--flex StyledProps--flex-distribution-space-between"
                >
                  <p
                    class="StyledProps--font StyledProps--main text StyledProps--color StyledProps--color-black StyledProps--font-weight-bold"
                  >
                    0
                  </p>
                </div>
              </div>
            </div>
            <div
              class="Layout--vertical StyledProps--main bottomSection"
            >
              <div
                class="Layout--vertical StyledProps--main StyledProps--margin-bottom-medium"
              >
                <p
                  class="StyledProps--font StyledProps--main StyledProps--font-weight-bold StyledProps--color StyledProps--color-grey600 StyledProps--margin-bottom-xsmall"
                >
                  cd.serviceDashboard.serviceInstances
                </p>
                <div
                  class="Layout--horizontal StyledProps--main StyledProps--flex StyledProps--flex-alignItems-center StyledProps--flex-distribution-space-between"
                >
                  <p
                    class="StyledProps--font StyledProps--main text StyledProps--color StyledProps--color-black StyledProps--font-weight-bold"
                  >
                    0
                  </p>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="bp3-card bp3-elevation-0 Card--card card"
        >
          <div
            class="bp3-card bp3-elevation-0 Card--card card"
          >
            <div
              class="Layout--vertical StyledProps--main"
              style="height: 100%;"
            >
              <p
                class="StyledProps--font StyledProps--main StyledProps--font-weight-bold StyledProps--color StyledProps--color-grey600"
              >
                cd.serviceDashboard.mostActiveServices
              </p>
              <div
                class="StyledProps--font StyledProps--main StyledProps--margin-bottom-xlarge"
              >
                <div
                  class="Layout--horizontal StyledProps--main"
                >
                  <p
                    class="StyledProps--font StyledProps--main environmentType StyledProps--font-size-small StyledProps--font-weight-semi-bold StyledProps--intent-primary"
                    data-test="mostActiveServicesWidgetEnvironmentType"
                  >
                    all
                  </p>
                  <p
                    class="StyledProps--font StyledProps--main environmentType StyledProps--font-size-small StyledProps--font-weight-semi-bold"
                    data-test="mostActiveServicesWidgetEnvironmentType"
                  >
                    cd.serviceDashboard.prod
                  </p>
                  <p
                    class="StyledProps--font StyledProps--main environmentType StyledProps--font-size-small StyledProps--font-weight-semi-bold"
                    data-test="mostActiveServicesWidgetEnvironmentType"
                  >
                    cd.serviceDashboard.nonProd
                  </p>
                </div>
              </div>
              <div
                class="Layout--horizontal StyledProps--main StyledProps--margin-bottom-large"
              >
                <div
                  class="typeContainer typeSelected"
                  data-test="mostActiveServicesWidgetTypeSelected"
                >
                  <p
                    class="StyledProps--font StyledProps--main StyledProps--font-size-xsmall StyledProps--font-weight-semi-bold StyledProps--color StyledProps--color-white"
                  >
                    deploymentsText
                  </p>
                </div>
                <div
                  class="typeContainer"
                  data-test="mostActiveServicesWidgetType"
                >
                  <p
                    class="StyledProps--font StyledProps--main StyledProps--font-size-xsmall StyledProps--font-weight-semi-bold StyledProps--color StyledProps--color-black"
                  >
                    errors
                  </p>
                </div>
              </div>
              <div
                class="Layout--vertical StyledProps--main mostActiveServicesWidgetEmpty StyledProps--flex StyledProps--flex-align-center-center"
                data-test="mostActiveServicesWidgetEmpty"
              >
                <div
                  class="StyledProps--font StyledProps--main StyledProps--margin-bottom-medium"
                >
                  <img
                    height="50"
                    src="test-file-stub"
                    style="align-self: center;"
                    width="50"
                  />
                </div>
                <p
                  class="StyledProps--font StyledProps--main StyledProps--color StyledProps--color-grey400"
                >
                  cd.serviceDashboard.noServiceInstances
                </p>
              </div>
            </div>
          </div>
          <div
            class="separator"
          />
          <div
            class="bp3-card bp3-elevation-0 Card--card card"
          >
            <div
              class="Layout--vertical StyledProps--main"
              style="height: 100%;"
            >
              <div
                class="Layout--vertical StyledProps--main StyledProps--flex StyledProps--flex-align-center-center"
                data-test="deploymentsWidgetEmpty"
                style="height: 100%;"
              >
                <img
                  height="100"
                  src="test-file-stub"
                  style="align-self: center;"
                  width="150"
                />
                <p
                  class="StyledProps--font StyledProps--main StyledProps--color StyledProps--color-grey400 StyledProps--margin-top-medium"
                >
                  cd.serviceDashboard.noDeployments
                </p>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="Layout--vertical StyledProps--main container"
      >
        <div
          class="Layout--horizontal StyledProps--main listHeader StyledProps--flex StyledProps--flex-distribution-space-between StyledProps--padding-top-medium StyledProps--padding-bottom-medium"
        >
          <p
            class="StyledProps--font StyledProps--main StyledProps--font-size-medium StyledProps--font-weight-semi-bold StyledProps--color StyledProps--color-grey700"
          >
            cd.serviceDashboard.totalServices
          </p>
          <div
            class="Layout--horizontal StyledProps--main StyledProps--margin-left-small"
          >
            <div
              class="bp3-input-group ui-search-box ExpandingSearchInput--main expandSearch ExpandingSearchInput--flip "
              data-name=""
              style="max-width: 200px;"
            >
              <span
                class="bp3-icon StyledProps--main ExpandingSearchInput--icon ExpandingSearchInput--flipicon"
                data-icon="thinner-search"
              >
                <svg
                  fill="none"
                  height="14"
                  viewBox="0 0 16 16"
                  width="14"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M7.125 13.25c1.617 0 3.32-.695 4.375-1.75L15 15l-3.5-3.5c1.021-1.005 1.75-2.894 1.75-4.375a6.125 6.125 0 10-6.125 6.125z"
                    stroke="currentColor"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                  />
                </svg>
              </span>
              <input
                class="bp3-input  ExpandingSearchInput--filter"
                dir="auto"
                placeholder="search"
                style="padding-right: 0px;"
                type="search"
                value=""
              />
            </div>
            <div
              class="bp3-popover-wrapper DropDown--main DropDown--disabled"
            >
              <div
                class="bp3-popover-target"
              >
                <div
                  class=""
                >
                  <div
                    class="Layout--horizontal StyledProps--main DropDown--dropdownButton StyledProps--flex"
                    data-testid="filter-select"
                    style="min-width: 220px;"
                  >
                    <div
                      class="Layout--horizontal StyledProps--main DropDown--labelWrapper StyledProps--flex"
                    >
                      <p
                        class="StyledProps--font StyledProps--main DropDown--label"
                        data-testid="dropdown-value"
                      >
                        common.filters.noFilterSaved
                      </p>
                    </div>
                    <div
                      class="Layout--horizontal StyledProps--main DropDown--btnWrapper StyledProps--flex"
                    >
                      <span
                        class="bp3-icon StyledProps--main StyledProps--color StyledProps--color-grey400"
                        data-icon="main-chevron-down"
                      >
                        <svg
                          height="8"
                          viewBox="0 0 14 8"
                          width="8"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            d="M13 0c-.3 0-.5.1-.7.3L7 5.6 1.7.3C1.5.1 1.3 0 1 0 .4 0 0 .4 0 1c0 .3.1.5.3.7l6 6c.2.2.4.3.7.3s.5-.1.7-.3l6-6c.2-.2.3-.4.3-.7 0-.6-.4-1-1-1z"
                            fill="#4d5969"
                          />
                        </svg>
                      </span>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="filterButtonContainer"
            >
              <button
                class="bp3-button bp3-minimal Button--button StyledProps--font StyledProps--main ngFilter StyledProps--intent-primary Button--with-current-color Button--without-shadow"
                id="ngfilterbtn"
                type="button"
              >
                <span
                  class="bp3-icon StyledProps--main"
                  data-icon="ng-filter"
                >
                  <svg
                    height="22"
                    viewBox="0 0 24 24"
                    width="22"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <mask
                      fill="#fff"
                      id="ng-filter_svg__a"
                    >
                      <path
                        clip-rule="evenodd"
                        d="M16.5 5a.5.5 0 00-1 0v1H5a.5.5 0 000 1h10.5v1a.5.5 0 001 0V7H19a.5.5 0 000-1h-2.5V5zm-12 7a.5.5 0 01.5-.5h3v-1a.5.5 0 011 0v1h10a.5.5 0 010 1H9v1a.5.5 0 01-1 0v-1H5a.5.5 0 01-.5-.5zm0 5.5A.5.5 0 015 17h7v-1a.5.5 0 011 0v1h6a.5.5 0 010 1h-6v1a.5.5 0 01-1 0v-1H5a.5.5 0 01-.5-.5z"
                        fill-rule="evenodd"
                      />
                    </mask>
                    <path
                      d="M15.5 6v1a1 1 0 001-1h-1zm0 1h1a1 1 0 00-1-1v1zm1 0V6a1 1 0 00-1 1h1zm0-1h-1a1 1 0 001 1V6zM8 11.5v1a1 1 0 001-1H8zm1 0H8a1 1 0 001 1v-1zm0 1v-1a1 1 0 00-1 1h1zm-1 0h1a1 1 0 00-1-1v1zm4 4.5v1a1 1 0 001-1h-1zm1 0h-1a1 1 0 001 1v-1zm0 1v-1a1 1 0 00-1 1h1zm-1 0h1a1 1 0 00-1-1v1zm4-12.5a.5.5 0 01-.5-.5h2A1.5 1.5 0 0016 3.5v2zm.5-.5a.5.5 0 01-.5.5v-2A1.5 1.5 0 0014.5 5h2zm0 1V5h-2v1h2zM5 7h10.5V5H5v2zm.5-.5A.5.5 0 015 7V5a1.5 1.5 0 00-1.5 1.5h2zM5 6a.5.5 0 01.5.5h-2A1.5 1.5 0 005 8V6zm10.5 0H5v2h10.5V6zm1 2V7h-2v1h2zm-.5-.5a.5.5 0 01.5.5h-2A1.5 1.5 0 0016 9.5v-2zm-.5.5a.5.5 0 01.5-.5v2A1.5 1.5 0 0017.5 8h-2zm0-1v1h2V7h-2zM19 6h-2.5v2H19V6zm-.5.5A.5.5 0 0119 6v2a1.5 1.5 0 001.5-1.5h-2zm.5.5a.5.5 0 01-.5-.5h2A1.5 1.5 0 0019 5v2zm-2.5 0H19V5h-2.5v2zm-1-2v1h2V5h-2zM5 10.5A1.5 1.5 0 003.5 12h2a.5.5 0 01-.5.5v-2zm3 0H5v2h3v-2zm-1 0v1h2v-1H7zM8.5 9A1.5 1.5 0 007 10.5h2a.5.5 0 01-.5.5V9zm1.5 1.5A1.5 1.5 0 008.5 9v2a.5.5 0 01-.5-.5h2zm0 1v-1H8v1h2zm9-1H9v2h10v-2zm1.5 1.5a1.5 1.5 0 00-1.5-1.5v2a.5.5 0 01-.5-.5h2zM19 13.5a1.5 1.5 0 001.5-1.5h-2a.5.5 0 01.5-.5v2zm-10 0h10v-2H9v2zm1 0v-1H8v1h2zM8.5 15a1.5 1.5 0 001.5-1.5H8a.5.5 0 01.5-.5v2zM7 13.5A1.5 1.5 0 008.5 15v-2a.5.5 0 01.5.5H7zm0-1v1h2v-1H7zm-2 1h3v-2H5v2zM3.5 12A1.5 1.5 0 005 13.5v-2a.5.5 0 01.5.5h-2zM5 16a1.5 1.5 0 00-1.5 1.5h2a.5.5 0 01-.5.5v-2zm7 0H5v2h7v-2zm1 1v-1h-2v1h2zm0-1a.5.5 0 01-.5.5v-2A1.5 1.5 0 0011 16h2zm-.5.5a.5.5 0 01-.5-.5h2a1.5 1.5 0 00-1.5-1.5v2zM12 16v1h2v-1h-2zm7 0h-6v2h6v-2zm1.5 1.5A1.5 1.5 0 0019 16v2a.5.5 0 01-.5-.5h2zM19 19a1.5 1.5 0 001.5-1.5h-2a.5.5 0 01.5-.5v2zm-6 0h6v-2h-6v2zm-1-1v1h2v-1h-2zm0 1a.5.5 0 01.5-.5v2A1.5 1.5 0 0014 19h-2zm.5-.5a.5.5 0 01.5.5h-2a1.5 1.5 0 001.5 1.5v-2zm.5.5v-1h-2v1h2zm-8 0h7v-2H5v2zm-1.5-1.5A1.5 1.5 0 005 19v-2a.5.5 0 01.5.5h-2z"
                      fill="#383946"
                      mask="url(#ng-filter_svg__a)"
                    />
                  </svg>
                </span>
              </button>
            </div>
            <div
              class="Layout--horizontal StyledProps--main"
            />
          </div>
        </div>
        <div
          class="Layout--vertical StyledProps--main StyledProps--flex StyledProps--flex-align-center-center"
          data-test="deploymentsWidgetEmpty"
          style="height: 80%;"
        >
          <img
            height="100"
            src="test-file-stub"
            style="align-self: center;"
            width="150"
          />
          <p
            class="StyledProps--font StyledProps--main StyledProps--color StyledProps--color-grey400 StyledProps--margin-top-medium"
          >
            cd.serviceDashboard.noServiceDetails
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
